examples/proj_create_example.c

```diff
@@ -0,0 +1,66 @@
+/*******************************************************************************
+    Simple example code demonstrating use of the proj.h API for 2D coordinate
+    transformations.
+
+    It illustrates the proj_create() function. It shows how, given a
+    projection, one can convert from geodetic to projected coordinates
+    and vice-versa.
+
+    Note: This file is in-lined in the documentation. Any changes must be
+    reflected in docs/source/development/quickstart.rst
+
+*******************************************************************************/
+#include <proj.h>
+#include <stdio.h>
+#include <math.h>
+
+int main (void) {
+  PJ_CONTEXT *C;
+  PJ *P;
+
+  /* Create the context. */
+  /* You may set C=PJ_DEFAULT_CTX if you are sure you will     */
+  /* use PJ objects from only one thread                       */
+  C = proj_context_create();
+
+  /* Create the projection. */
+  P = proj_create(C, "+proj=utm +zone=32 +datum=WGS84");
```

Comment:
This doesn't reflect the discussion in the PR. Please adjust according to @rouault's suggestion https://github.com/OSGeo/PROJ/pull/3515#issuecomment-1369250539

Once the pipeline is changed adjustments are needed in the following code, as we are now transforming between two projected CRS's. Comments inlined where I believe changes are necessary.

Reply:
> This doesn't reflect the discussion in the PR. 

When it comes to @rouault's suggestion, I don't think the user needs such an awfully long pipeline string to be able to understand how to use this simple function. That suggestion was meant to suggest to you that proj_create is not as simple as you think. Likely it won't help the user looking for basic functionality. 

I am happy enough to accept any suggestions you put in. You are welcome to adopt them and pull in that example. Thank you.